### ISSUE TITLE
resize with pad transformation

### ISSUE THREAD

**Issue by oekosheri** [pytorch/vision#6236]
### 🚀 The feature

In tensorflow tf.image has a method, tf.image.resize_with_pad, that pads and resizes if the aspect ratio of input and output images are different to avoid distortion. I couldn't find an equivalent in torch transformations and had to write it myself. I think it would be a useful feature to have.  

### Motivation, pitch

When moving to pytoch from Tensorflow, one does not want to lose handy features!

**Comment by zhiqwang** [comment]
Hi @oekosheri , you can check following function, it will do bottom-right padding mode

https://github.com/pytorch/vision/blob/d6e39ff76c82c7510f68a7aa637f015e7a86f217/torchvision/models/detection/transform.py#L25-L71

And I wrote a similar wrap-around letterboxing mode at belows:

https://github.com/zhiqwang/yolov5-rt-stack/blob/main/yolort/models/transform.py#L65-L109

**Comment by oekosheri** [comment]
Hi @zhiqwang, Thanks! you mean I can use "TORCH.NN.FUNCTIONAL.INTERPOLATE" ? I tried it on a an image tensor now and it constantly gives value error of input/output size not matching. 
Also, this is pretty hidden. Why not add a simple wrapper to resize that does padding when aspect ratio can't be preserved?

**Comment by zhiqwang** [comment]
> Also, this is pretty hidden. Why not add a simple wrapper to resize that does padding when aspect ratio can't be preserved?

Let's invite @datumbox to this disscusion, and hear his viewpoint on this problem.

**Comment by zhiqwang** [comment]
Just FYI, a previous issue #3286 also has some relevance to the discussion here.

**Comment by oekosheri** [comment]
@datumbox They are padded to ensure images that have different original aspect ratio to the final one, don't get distorted. Distorted images may not work well with CNNs. I updated a mistake in the code above. As it is now, it produces the exact output that tf.image.resize_with_pad does.

**Comment by datumbox** [comment]
@oekosheri Thanks for the references and context. I'll sync with @vfdev-5 offline to see if we can add this on the new API and how. I'll leave the issue open to ensure it stays on our radar.

**Comment by Inkorak** [comment]
Yes, I also think that such a transformation would be very useful. I also had cases when images of different resolutions and aspect ratios, but when cropping images, I could lose pieces important for classification (this was a classification of defects and they could be on the edge of the image) and I would like to maintain the aspect ratio in order to avoid strong distortions. So I had to use a combination of ongestMaxSize and PadIfNeeded from the albumentations library. I would like something similar, you can implement it as suggested here in the form of one transformation.

**Comment by H-Sorkatti** [comment]
I Strongly second this feature. It is very a important transformation to have.
I almost always have to use hacks to resolve this when working with images.

**Comment by AsiaCao** [comment]
ditto, it'd be really handy to have one.
Our team also uses such a feature. We currently implemented a custom version with albumentations that only works for numpy (not torch tensor). And we are looking for an alternative that works with torch tensor and can be converted/embedded into an onnx graph via `torch.onnx.export`.

curious, do you think the torchvision team could implement it soon? @datumbox @zhiqwang

**Comment by datumbox** [comment]
@AsiaCao Thanks for the input. Right now we are focusing on finalizing the Transforms V2 API. Once we complete the work on that front, we can review this request and see what's the best way forwards.

**Comment by AsiaCao** [comment]
thanks @datumbox

**Comment by swap-10** [comment]
Any plans for this to be implemented now? This would be convenient to have.
Thanks!

**Comment by amanikiruga** [comment]
This
